src/worker/mod.rs

```diff
@@ -34,14 +34,18 @@ pub enum PreferWorkerKind {
     Multithreaded,
 }
 
+
 /// Execute something with a worker system.
 pub fn with_worker<T>(prefer: PreferWorkerKind, f: impl FnOnce(&mut dyn Worker) -> T) -> T {
+    #[cfg(any(target_arch = "asmjs", target_arch = "wasm32"))]
+    {
+        return self::immediate::with_immediate(f);
+    }
     match prefer {
-        #[cfg(not(any(target_arch = "asmjs", target_arch = "wasm32")))]
         #[cfg(feature = "rayon")]
```

Comment:
Would replacing this with `#[cfg(all(not(any(target_arch = "asmjs", target_arch = "wasm32")), feature = "rayon"))]` work instead?

Reply:
Yep, it's the same 😊
It is the "double line" cfg the real issue

Reply:
Could we do that instead then? Seems nicer than duplicating the self::immediate::with_immediate() call.